src/common/associatedToken.ts

```diff
@@ -209,4 +273,30 @@ export class AssociatedToken {
     const ix = Token.createCloseAccountInstruction(TOKEN_PROGRAM_ID, tokenAddress, rentDestination, owner, multiSigner)
     instructions.push(ix)
   }
+
+  //create IX for wrapping SOL
+  static async withWrapIfNativeMint(
+    instructions: TransactionInstruction[],
+    provider: Provider,
+    owner: PublicKey,
+    mint: PublicKey,
+    amount: BN
+  ) {
+    //only run if mint is wrapped sol mint
+    if (mint.equals(NATIVE_MINT)) {
+      //this will add instructions to create ata if ata does not exist, if exist, we will get the ata address
+      const ata = await this.withCreate(instructions, provider, owner, mint)
+      //IX to transfer sol to ATA
+      const transferIx = SystemProgram.transfer({
+        fromPubkey: owner,
+        //fix? may be imprecise?
+        lamports: amount.toNumber(),
```

Comment:
Correct it's imprecise. Best to be precise but if it causes trouble we can write a transfer ix that takes a BN 

You will want to use bnToNumber in the case of it actually being an imprecise number (>2^53) it returns an imprecise number instead of throwing an error
```suggestion
        //fix? may be imprecise?
        lamports: bnToNumber(amount),
```